### DISCUSSION TITLE
xunit results missing some fields

### DISCUSSION THREAD

**Discussion by luckyh** [avocado-framework/avocado#5330] [Q&A] [answered]
This is one of the issues we discovered when we were going to switch to nrunner for the internal use of Avocado-VT test execution. From the following comparison we can see `testcase.classname`, `error.type`(and also `failure.type`) and `error.description`(and also `failure.description`) are available in the result of legacy runner, but missing in the one of nrunner (whereas there are only the strings `<unknown>` as the placeholder).

**legacy runner**
```xml
<testcase classname="VirtTest" name="11-Host_RHEL.m8.u5.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_stream.q35" time="324.873">
  <error type="VMIPAddressMissingError" message="No ipv4 DHCP lease for MAC 9a:d5:04:53:4c:60">
    Traceback (most recent call last): File "/usr/local/lib/python3.9/site-packages/avocado_framework-92.0-py3.9.egg/avocado/core/test.py" ...
  </error>
  ...
</testcase>
```
**nrunner**
```xml
<testcase classname="<unknown>" name="Host_RHEL.m8.u5.v0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35" time="101.714">
  <error type="<unknown>" message="Qemu crashed: ..."><unknown></error>
  ...
</testcase>
```

After checking the code, it seemed that this information had not been passed via tests discovery/messages when executing tests with nrunner. So I would like to confirm that if it is the desired behavior and for what purpose, or else, is it considered to be a bug?

**Comment by beraldoleal** [comment, marked as answer]
Hi @luckyh, thanks for reporting. I just created an issue from this discussion.

**Reply by luckyh** [reply]
ok, thank you!
